pkg/process/monitor/process_monitor.go

```diff
@@ -267,6 +276,8 @@ func (pm *ProcessMonitor) Initialize() error {
 					return
 				}
 				log.Errorf("process monitor error: %s", err)
+				pm.terminateProcessMonitor()
+				terminated = true
 				pm.Stop()
```

Comment:
replacing 
```
  pm.Stop()
  return
```
with
`  close(pm.done)`

will do the fix without adding more complexity

Reply:
disagree, we should have a single location to "clean" the state, that duplication will cause bugs and leaks because we remember to modify only the obvious location.

Reply:
https://github.com/DataDog/datadog-agent/pull/16309

Reply:
that's not how we should work here, if there is a disagreement we shouldn't write a new PR instead... 

and for your PR, if stop is being called twice we'll panic. that's the only "benefit" of calling the original Stop function.

Reply:
I push a PR as it's more simple to show the diff (I should had move it as draft)
I did push the last version resting refcount

Your PR is correct but add extra complexity ... merge it